kaggle_environments/core.py

```diff
@@ -538,27 +551,33 @@ def __process_specification(self, spec):
     def __agent_runner(self, agents):
         # Generate the agents.
         agents = [
-            Agent(a, self.configuration, self)
-            if a is not None
+            Agent(agent, self)
+            if agent is not None
             else None
-            for a in agents
+            for agent in agents
         ]
 
         def act(none_action=None):
             if len(agents) != len(self.state):
                 raise InvalidArgument(
                     "Number of agents must match the state length")
 
-            actions = [0] * len(agents)
-            for i, agent in enumerate(agents):
-                if self.state[i]["status"] != "ACTIVE":
-                    actions[i] = None
-                elif agent is None:
-                    actions[i] = none_action
-                else:
-                    state = self.__get_shared_state(i)
-                    actions[i] = agent.act(state["observation"])
-            return actions
+            act_args = [
+                (
+                    agent,
+                    self.__get_shared_state(i),
+                    self.configuration,
+                    none_action,
+                )
+                for i, agent in enumerate(agents)
+            ]
+
+            if all(agent.is_picklable for agent in agents):
```

Comment:
If any agent is not picklable we don't parallelize at all.

Reply:
nit: Again agree that `should_parallelize` (or similar) makes this clearer